src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -328,7 +328,7 @@ bool PDPServer2::server_update_routine()
     bool result = process_writers_acknowledgements();  // server + ddb(functor_with_ddb)
     // result |= process_data_queue();                    // all ddb
     result |= process_disposals();                     // server + ddb(get_disposals, clear_changes_to_disposes)
-    // result |= process_dirty_topics();                  // all ddb
+    result |= discovery_db_.process_dirty_topics();    // all ddb
```

Comment:
```suggestion
    result |= process_dirty_topics();    // all ddb
```
I would suggest to create a new method in PDPServer2. It is not necessary but keeps the logic uniform along the process methods

Reply:
:+1: